### DISCUSSION TITLE
How to create frame around canvas?

### DISCUSSION THREAD

**Discussion by paulm17** [fabricjs/fabric.js#8099] [Q&A] [answered]
I'm trying to create a similar effect to the following: https://codepen.io/jdhaliwal/pen/wvKpBje

To further sets of features I'm looking for is to change the stroke width and color.

I've been looking at a lot of other issues and discussions on how to solve this issue, but I'm confused as to the best approach.

Do I:

A) Do I create path objects and try to create a non-selectable frame.  

B) Or do I add a transparent image with a border to the canvas background image?  Although I'm not sure this is possible.

Has anyone achieved this?  Thanks

**Comment by xorb** [comment]
Looking for something like this?
https://github.com/layerhub-io/react-editor

**Comment by ShaMan123** [comment, marked as answer]
using backgroundImage is a good approach
It doesn't need to be an image, set it to a rect (bad naming)
Look at #7901

**Reply by paulm17** [reply]
Thanks, I will definitely take a look at that.  I'm working on something else with the image editor, but I'll update this discussion soon!

Fyi, I'm wanting to replicate this:

https://www.loom.com/share/eec95e5d6e374a20b3e90d14ec57eb2c
